OpenRA.Mods.Common/Traits/CreatesShroud.cs

```diff
@@ -21,12 +23,21 @@ public class CreatesShroudInfo : AffectsShroudInfo
 		public override object Create(ActorInitializer init) { return new CreatesShroud(init.Self, this); }
 	}
 
-	public class CreatesShroud : AffectsShroud
+	public class CreatesShroud : AffectsShroud, INotifyCreated
 	{
 		readonly CreatesShroudInfo info;
+		IEnumerable<ICreatesShroudModifier> rangeModifiers;
 
 		public CreatesShroud(Actor self, CreatesShroudInfo info)
-			: base(self, info) { this.info = info; }
+			: base(self, info)
+		{
+			this.info = info;
+		}
+
+		void INotifyCreated.Created(Actor self)
```

Comment:
This is a conditional trait, so implementing `INotifyCreated` here will break things!

Replace this with
```csharp
protected override void Created(Actor self)
{
	base.Created(self);
	rangeModifiers = self.TraitsImplementing<ICreatesShroudModifier>().ToArray();
}
```

and drop the interface above